src/plugins/action_generate_review_assignment_records/src/generateReviewAssignments.ts

```diff
@@ -1,112 +1,105 @@
+      const isLastLevel = nextReviewLevel === numReviewLevels
+      return generateNextReviewAssignments({
+        db,
+        applicationId,
+        templateId,
+        nextReviewLevel,
+        nextStageNumber: stageNumber,
+        nextStageId: stageId,
+        isLastLevel,
+      })
     }
-    const nextReviewLevel = currentReviewLevel + 1
-
-    const nextLevelReviewers = await db.getPersonnelForApplicationStageLevel(
-      templateId,
-      stageNumber,
-      nextReviewLevel,
-      'Review'
-    )
-
-    const reviewAssignments: ReviewAssignmentObject = {}
-
-    // Build reviewers into object map so we can combine duplicate user_orgs
-    // and merge their section code restrictions
-    nextLevelReviewers.forEach((reviewer: Reviewer) => {
-      const { userId, orgId, restrictions } = reviewer
-
-      const templateSectionRestrictions = restrictions
-        ? restrictions?.templateSectionRestrictions
-        : null
-
-      const userOrgKey = `${userId}_${orgId ? orgId : 0}`
-      if (reviewAssignments[userOrgKey])
-        reviewAssignments[userOrgKey].templateSectionRestrictions = mergeSectionRestrictions(
-          reviewAssignments[userOrgKey].templateSectionRestrictions,
-          templateSectionRestrictions
-        )
-      else
-        reviewAssignments[userOrgKey] = {
-          reviewerId: userId,
-          orgId,
-          stageId,
-          stageNumber,
-          // TO-DO: allow STATUS to be configurable in template
-          status: nextReviewLevel === 1 ? AssignmentStatus.AVAILABLE : AssignmentStatus.SELF_ASSIGN,
+    // For level 1+ or next stages review assignment
+    // TODO - #440: Needs to lock same level reviews
+    // TODO: Unlock lower level review when consolidator has submitted "Change requested"
+    else {
+      const {
+        stageNumber: previousStage,
+        level: previousLevel,
+      } = await DBConnect.getCurrentReviewLevel(reviewId)
+      console.log('Review existing', previousStage, previousLevel)
+      // Review in new stage - first level
+      if (previousStage !== stageNumber) {
+        if (numReviewLevels === 0) {
+          console.log(
+            `No reviewer with level associated to stageNumber ${stageNumber}, no review assignments to generate.`
+          )
+          return {
+            status: 'Success',
+            error_log: '',
+            output: {},
+          }
+        }
+        const nextReviewLevel = 1
+        const isLastLevel = nextReviewLevel === numReviewLevels
+        console.log('New stage - total levels: ', numReviewLevels, stageNumber, previousStage)
+        return generateNextReviewAssignments({
+          db,
           applicationId,
-          templateSectionRestrictions,
-          level: nextReviewLevel,
-          isLastLevel: nextReviewLevel === numReviewLevels,
+          templateId,
+          nextReviewLevel,
+          nextStageNumber: stageNumber,
+          nextStageId: stageId,
+          isLastLevel,
+        })
```

Comment:
Same here -- can the return block go outside all the conditionals, since it's the same for all three?

Reply:
Happy to see suggestions for how to move this block, but it looks ok to me like this for now

Reply:
Ok. I think yesterday was just a bad day for me - realised it can be simplified a lot by this suggestion. Thanks : ) Done: 1cc6108